src/api/v1alpha1/component.go

```diff
@@ -124,6 +124,10 @@ type ZarfFile struct {
 	Symlinks []string `json:"symlinks,omitempty"`
 	// Local folder or file to be extracted from a 'source' archive.
 	ExtractPath string `json:"extractPath,omitempty"`
+	// [alpha]
+	// Template enables go-templates inside manifests. This is useful for parameterizing fields that the value will be
+	// known at deploy-time. See documentation for Zarf Values for how to set these values.
+	Template bool `json:"template,omitempty"`
```

Comment:
One strategy I've liked when using a boolean is having a boolptr and a function like `isTemplate` or `shouldTemplate` here and for manifests. This way instead of the default always be false, we can control the default. For example, with manifests in the next schema version, they will be gotemplatable by default. 

We do something similar with the required field and [isRequired](https://github.com/zarf-dev/zarf/blob/07c7c7cc0dd242e367b3aa7df91d7c67863143d2/src/api/v1alpha1/component.go#L87). This allows the object to own what the default of the value should be when a bool is used

Reply:
This tracks - updated in latest and defaulting to false